src/PaymentGateways/PayPalCommerce/PayPalOrder.php

```diff
@@ -0,0 +1,84 @@
+<?php
+namespace Give\PaymentGateways\PayPalCommerce;
+
+use Give\Helpers\ArrayDataSet;
+use InvalidArgumentException;
+use stdClass;
+
+/**
+ * Class PayPalOrder
+ * @package Give\PaymentGateways\PayPalCommerce
+ *
+ * @property-read string $id
+ * @property-read string $intent
+ * @property-read stdClass $purchaseUnit
+ * @property-read stdClass $payer
+ * @property-read string $createTime
+ * @property-read string $updateTime
+ * @property-read string $links
+ * @property-read string $status
+ */
+class PayPalOrder {
+	/**
+	 * Create PayPalOrder object from given array.
+	 *
+	 * @since 2.8.0
+	 *
+	 * @param $array
+	 *
+	 * @return PayPalOrder
+	 */
+	public static function fromArray( $array ) {
+		$order = new static();
+
+		$order->validate( $array );
+		$array = ArrayDataSet::camelCaseKeys( $array );
+
+		foreach ( $array as $itemName => $value ) {
+			if ( 'purchaseUnits' === $itemName ) {
+				$value = current( $value );
+			}
+
+			$order->{$itemName} = $value;
```

Comment:
Please avoid dynamic properties. The fact that you're setting things this way is fine, but make sure that the properties are defined in the class as well.

Reply:
https://github.com/impress-org/givewp/compare/f549508bd6a0...ae1fb915985b